File: python/字符串/293. 翻转游戏.py
from typing import List
class Solution:
    def generatePossibleNextMoves(self, s: str) -> List[str]:
        i = 0
        res = []
        while i < len(s)-1:
            tmp = list(s)
            if s[i] == '+' and s[i+1] == '+':
                tmp[i]= '-'
                tmp[i+1] = '-'
                res.append(''.join(tmp))
            i+=1
        return res
s = Solution()
test = "++++"
print(s.generatePossibleNextMoves(test))

File: go/string/293. 翻转游戏.go
package string

func generatePossibleNextMoves(s string) []string {
	var res []string
	for i := 0; i < len(s)-1; i++ {
		tmp := []byte(s)
		if s[i] == '+' && s[i+1] == '+' {
			tmp[i] = '-'
			tmp[i+1] = '-'
			res = append(res, string(tmp))
			continue
		}
	}
	return res
}
